opm/autodiff/AquiferFetkovich.hpp

```diff
@@ -0,0 +1,387 @@
+            // Do not make the connection if the product of the two cellIdx > 0. This is because the
+            // face is within the reservoir/not connected to boundary. (We still have yet to check for inactive cell adjoining)
+            double faceArea = 0.;
+            const auto cellNeighbour0 = faceCells(faceIdx,0);
+            const auto cellNeighbour1 = faceCells(faceIdx,1);
+            const auto defaultFaceArea = Opm::UgGridHelpers::faceArea(ugrid, faceIdx);
+            const auto calculatedFaceArea = (!connection.influx_coeff.at(idx))?
+            defaultFaceArea :
+            *(connection.influx_coeff.at(idx));
+            faceArea = (cellNeighbour0 * cellNeighbour1 > 0)? 0. : calculatedFaceArea;
+            if (cellNeighbour1 == 0){
+              faceArea = (cellNeighbour0 < 0)? faceArea : 0.;
+            }
+            else if (cellNeighbour0 == 0){
+              faceArea = (cellNeighbour1 < 0)? faceArea : 0.;
+            }
+            return faceArea;
+          }
+
+          // This function is used to initialize and calculate the alpha_i for each grid connection to the aquifer
+          inline void initializeConnections(const Aquancon::AquanconOutput& connection)
+          {
+            const auto& eclState = ebos_simulator_.vanguard().eclState();
+            const auto& ugrid = ebos_simulator_.vanguard().grid();
+            const auto& grid = eclState.getInputGrid();
+
+            cell_idx_ = connection.global_index;
+            auto globalCellIdx = ugrid.globalCell();
+
+            assert( cell_idx_ == connection.global_index);
+            assert( (cell_idx_.size() == connection.influx_coeff.size()) );
+            assert( (connection.influx_coeff.size() == connection.influx_multiplier.size()) );
+            assert( (connection.influx_multiplier.size() == connection.reservoir_face_dir.size()) );
+
+            // We hack the cell depth values for now. We can actually get it from elementcontext pos
+            cell_depth_.resize(cell_idx_.size(), aqufetp_data_.d0);
+            alphai_.resize(cell_idx_.size(), 1.0);
+            faceArea_connected_.resize(cell_idx_.size(),0.0);
+
+            auto cell2Faces = Opm::UgGridHelpers::cell2Faces(ugrid);
+            auto faceCells  = Opm::UgGridHelpers::faceCells(ugrid);
+
+            // Translate the C face tag into the enum used by opm-parser's TransMult class
+            Opm::FaceDir::DirEnum faceDirection;
+
+            // denom_face_areas is the sum of the areas connected to an aquifer
+            Scalar denom_face_areas = 0.;
+            cellToConnectionIdx_.resize(ebos_simulator_.gridView().size(/*codim=*/0), -1);
+            for (size_t idx = 0; idx < cell_idx_.size(); ++idx)
+            {
+              cellToConnectionIdx_[cell_idx_[idx]] = idx;
+              const int cell_index = cartesian_to_compressed_.at(cell_idx_[idx]);
+              auto cellFacesRange = cell2Faces[cell_index];

```

Comment:
The above code should be something as follows, 
```
                    const int cell_index = cartesian_to_compressed_.at(cell_idx_[idx]);
                    cellToConnectionIdx_[cell_index] = idx;
                    const auto cellFacesRange = cell2Faces[cell_index];
```

Reply:
Changed!